modules/navigation/nav_link.cpp

```diff
@@ -94,6 +94,46 @@ void NavLink::set_end_position(const Vector3 p_position) {
 	request_sync();
 }
 
+void NavLink::set_navigation_layers(uint32_t p_navigation_layers) {
+	if (navigation_layers == p_navigation_layers) {
+		return;
+	}
+	navigation_layers = p_navigation_layers;
+	link_dirty = true;
+
+	request_sync();
+}
+
+void NavLink::set_enter_cost(real_t p_enter_cost) {
+	if (enter_cost == p_enter_cost) {
+		return;
+	}
+	enter_cost = MAX(p_enter_cost, 0.0);
```

Comment:
You should probably clamp the incoming cost before the if. As if your enter_cost is 0 and p_enter_cost is negative, you will sync for no reason